src/cplscheme/config/CouplingSchemeConfiguration.cpp

```diff
@@ -259,22 +259,19 @@ void CouplingSchemeConfiguration::xmlTagCallback(
     bool          initialize          = tag.getBooleanAttributeValue(ATTR_INITIALIZE);
     mesh::PtrData exchangeData;
     mesh::PtrMesh exchangeMesh;
-    for (const mesh::PtrMesh &mesh : _meshConfig->meshes()) {
-      if (mesh->getName() == nameMesh) {
-        for (const mesh::PtrData &data : mesh->data()) {
-          if (data->getName() == nameData) {
-            exchangeData = data;
-            exchangeMesh = mesh;
-            break;
-          }
-        }
-      }
-    }
-    PRECICE_CHECK(exchangeData.get(),
+
+    PRECICE_CHECK(_meshConfig->hasMeshName(nameMesh) && _meshConfig->getMesh(nameMesh)->hasDataName(nameData),
                   "Mesh \"{}\" with data \"{}\" not defined. "
                   "Please check the <exchange data=\"{}\" mesh=\"{}\" from=\"{}\" to=\"{}\" /> "
                   "tag in the <coupling-scheme:... /> of your precice-config.xml.",
                   nameMesh, nameData, nameData, nameMesh, nameParticipantFrom, nameParticipantTo);
+
+    const mesh::PtrMesh &mesh = _meshConfig->getMesh(nameMesh);
+    const mesh::PtrData &data = mesh->data(nameData);
+    PRECICE_ASSERT(mesh && data);
+    exchangeData = data;
+    exchangeMesh = mesh;
```

Comment:
Couldn't you merge the declaration and the initialization here (see line 260)

Reply:
True, we can still assert `exchangeData && exchangeMesh`

Reply:
Done in dec11c0af23f056ffa3f3eb3a54c8380accbcb5c